src/features/hotspots/root/HotspotsView.tsx

```diff
@@ -105,66 +154,60 @@ const HotspotsView = ({
   }, [hasHotspots, networkHotspots])
 
   useEffect(() => {
-    if (listIsDismissed && !selectedHotspot) {
+    if (viewState === 'list' && !selectedHotspot) {
       if (ownedHotspots && ownedHotspots.length > 0) {
         setSelectedHotspot(ownedHotspots[0])
       } else {
         focusClosestHotspot()
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [listIsDismissed, networkHotspots])
-
-  useEffect(() => {
-    if (startOnMap) {
```

Comment:
Some of this was for when you navigate here from the "view hotspots around me" so that it selects the closest hotspot to you and shows it first. Did you change how that works?

Reply:
Hmm it seems like that is no longer working on Android, but it seems related to previous location changes

Reply:
My intention wasn't to change how it works, just to bring all of that logic into one place. As far as I can tell everything is working as it was.

Reply:
I updated it so that when you dismiss the list it will bring the user to their location if they give permission otherwise it will bring them to one of their hotspot's locations.